pandas/types/cast.py

```diff
@@ -361,6 +377,11 @@ def _infer_dtype_from_scalar(val):
     elif is_complex(val):
         dtype = np.complex_
 
+    elif pandas_dtype:
```

Comment:
why is this parameter necessary, why wouldn't we *always* want to infer a pandas dtype if its there (e.g. ``Period``)?